lib/check.js

```diff
@@ -68,11 +68,12 @@ module.exports = function (options, callback) {
     options = {};
   }
 
-  options.proxy = options.proxy || Conf.proxy;
-  options.proxy = options.proxy || process.env.https_proxy || process.env.HTTPS_PROXY;
-  if (options.proxy) {
-    Conf.api.agent = new ProxyAgent(options.proxy);
-    delete options.proxy;
+  var proxy = options.proxy || Conf.proxy;
+  proxy = proxy || process.env.https_proxy || process.env.HTTPS_PROXY;
```

Comment:
it looks like it was this way to simplify coverage for tests. i left it so i didn't have to add even more tests to facilitate getting this change out quickly.